Source/PeleLMeX_DiffusionOp.cpp

```diff
@@ -146,26 +146,23 @@ DiffusionOp::diffuse_scalar(
     phi.emplace_back(
       a_phi[lev]->boxArray(), a_phi[lev]->DistributionMap(), ncomp, 1,
       amrex::MFInfo(), a_phi[lev]->Factory());
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
-#endif
-    for (amrex::MFIter mfi(phi[lev], amrex::TilingIfNotGPU()); mfi.isValid();
-         ++mfi) {
-      const amrex::Box& gbx = mfi.growntilebox();
-      auto const& a_phi_arr = a_phi[lev]->const_array(mfi, phi_comp);
-      auto const& a_rho_arr =
-        (have_density) != 0 ? a_density[lev]->const_array(mfi)
-                            : a_phi[lev]->const_array(
-                                mfi); // Get dummy amrex::Array4 if no density
-      auto const& phi_arr = phi[lev].array(mfi);
+    if (have_density == 0) {
+      amrex::MultiFab::Copy(
+        phi[lev], *a_phi[lev], phi_comp, 0, ncomp, phi[lev].nGrowVect());
+    } else {
+      auto const& a_phi_ma = a_phi[lev]->const_arrays();
+      auto const& a_rho_ma = a_density[lev]->const_arrays();
+      auto const& phi_ma = phi[lev].arrays();
       amrex::ParallelFor(
-        gbx, ncomp, [=] AMREX_GPU_DEVICE(int i, int j, int k, int n) noexcept {
-          if (have_density != 0) {
-            phi_arr(i, j, k, n) = a_phi_arr(i, j, k, n) / a_rho_arr(i, j, k);
-          } else {
-            phi_arr(i, j, k, n) = a_phi_arr(i, j, k, n);
-          }
+        phi[lev], phi[lev].nGrowVect(), ncomp,
```

Comment:
Replace with a Multifab divide call, similar to the multifab copy above? Same for other instances below in this file.